src/lib.rs

```diff
@@ -210,5 +221,34 @@ pub mod pallet {
             )
             .map_err(|e| format!("error in get_resource: {:?}", e).into())
         }
+
+        /// Get balance of given account in native currency converted to u128
+        pub fn get_balance(of: T::AccountId) -> Result<u128, Error<T>> {
+            let encoded_balance = T::Currency::free_balance(&of).encode();
+            if encoded_balance.len().ne(&16usize) {
+                return Err(Error::BalanceConversionFailed);
+            }
+            Ok(u128::from_be_bytes(
+                array_ref!(encoded_balance, 0, 16).to_owned(),
+            ))
+        }
+
+        // Get balance of given Move account in native currecy converted to u128
+        pub fn get_move_balance(of: &AccountAddress) -> Result<u128, Error<T>> {
+            Self::get_balance(Self::move_to_native(of)?)
+        }
+
+        // Transparent conversion move -> native
+        pub fn move_to_native(of: &AccountAddress) -> Result<T::AccountId, Error<T>> {
```

Comment:
Pallet already has conversions between Move<->Substrate addresses. Take a look here:
[https://github.com/eigerco/pallet-move/blob/main/src/address.rs](https://github.com/eigerco/pallet-move/blob/main/src/address.rs)

So, we should choose one :) Either use those from the link above or replace them with the new ones if we think they are better 👍 

`move_to_native` function is almost equivalent of `to_substrate_account` from address.rs.